Add VCF docs

tomwhite opened the pull request:
First (draft) attempt to include VCF docs for https://github.com/pystatgen/sgkit-vcf.

The API docs are included, but I needed to install the sgkit-vcf package ~~(with `--no-deps`, to avoid having to install cyvcf2, which is not needed for API signatures)~~.

![sgkit-vcf](https://user-images.githubusercontent.com/85085/92397853-67719a80-f11f-11ea-9703-9b4ed9a35b08.png)


Refs #54

jeromekelleher reviewed (approved):
LGTM @tomwhite. I wondered about the organisation a little bit, but I think this is quite logical. We'll probably want a subsection for "file formats" somewhere summarising sgkit's file format support, but that can come later.

tomwhite commented:
Thanks @jeromekelleher. I was going to write some more content for VCF before merging, but the organisation could be changed when we add BGEN and PLINK sections.

Are we happy with the build changes for the Sphinx docs? (Basically we have to install sgkit-vcf and all its dependencies.)

jeromekelleher commented:
> Are we happy with the build changes for the Sphinx docs? (Basically we have to install sgkit-vcf and all its dependencies.)

Yeah, I don't think that can be avoided. It's a big advantage of building the docs ourselves that we can just install the libs we need rather than having to mock stuff out like you do on rtd.

tomwhite commented:
I've updated this PR now that we've moved the VCF into the main sgkit repo (#289).

The changes are mirrored at http://tomwhite.github.com/sgkit/ to make them easier to review.

eric-czech reviewed (approved):
LGTM, great write-up @tomwhite!

The organization is a little weird to me too but perhaps the User Guide IO section becomes about what is common amongst all supported file formats -- how they have optional dependencies, the fact they shouldn't be used directly in favor of converting to zarr, which ones support cloud data and which don't, etc.
